Feat/comment

LeeJaeHyung opened the pull request:
## 📝 작업내용
-  댓글  CRUD 작업 + 댓글 좋아요 CRD

<br>

## ✨ 변경 사항
- [x] 새로운 기능 추가
- [ ] 버그 수정
- [ ] CSS 등 사용자 UI 디자인 변경
- [ ] 코드에 영향을 주지 않는 변경사항(오타 수정, 탭 사이즈 변경, 변수명 변경)
- [ ] 코드 리팩토링
- [ ] 주석 추가 및 수정
- [ ] 문서 수정
- [ ] 테스트 추가, 테스트 리팩토링
- [ ] 빌드 부분 혹은 패키지 매니저 수정
- [ ] 파일 혹은 폴더명 수정
- [ ] 파일 혹은 폴더 삭제
<br>

## To Reviewrs 🙏
-

LeeJaeHyung committed:
:sparkles:
댓글 좋아요 기능 생성

CommentController 수정, CommentService 수정, CommentLikeRepository 작성

LeeJaeHyung committed:
:twisted_rightwards_arrows: dev -> feat/comment
카카오 로그인 관련 가져와 merge

LeeJaeHyung committed:
:sparkles: comment 관련 swagger 추가
CommentController 수정

LeeJaeHyung committed:
Merge branch 'dev' into feat/comment

KIM-TABLE-NEXT reviewed (approved):
확인했습니다. 고생하셨습니다!

Esunsin commented on src/main/java/com/sparta/hmpah/controller/CommentController.java:
```diff
@@ -1,15 +1,83 @@
 @RestController
 @Slf4j
 @RequiredArgsConstructor
 @RequestMapping("/comment")
```
rest한 url을 위해 /comments가 어떠신가요?

Esunsin commented on src/main/java/com/sparta/hmpah/controller/CommentController.java:
```diff
@@ -1,15 +1,83 @@
+
+  private final CommentService commentService;
+
+  @GetMapping("/get-postId")
```
@RequsetParam으로 받으시면 url 없어도 될거 같습니다.

Esunsin commented on src/main/java/com/sparta/hmpah/controller/CommentController.java:
```diff
@@ -1,15 +1,83 @@
+    return commentService.getComments(postId);
+  }
+
+  @PostMapping("/create")//댓글 생성
```
method로 동사를 표현하기 떄문에 restful 관점에서 create는 없어도 될거 같습니다.
like의 url도 마찬가지로 method가 동사를 표현하여 likes로 바꾸시면 좋을거 같습니다

noeyodeel reviewed (approved):
확인했습니다! 수고하셨습니다 👏

Esunsin commented on src/main/java/com/sparta/hmpah/service/CommentService.java:
```diff
@@ -1,15 +1,140 @@
+    List<Comment> commentList = commentRepository.findByPostIdOrderByCreatedAtAsc(
+        postId);//.stream().map(CommentResponse::new).toList();
+    for (Comment m : commentList) {
+      System.out.println("m.getUser().toString() = " + m.getUser().toString());
```
log로 찍으시는 것이 어떤지 궁금합니다.

Esunsin commented on src/main/java/com/sparta/hmpah/service/CommentService.java:
```diff
@@ -1,15 +1,140 @@
+    return commentList.stream().map(CommentResponse::new).toList();
+  }
+
+  public CommentResponse createComment(CommentRequest requestDto,
```
파라미터값으로 userDetails 받는 것이 아닌 User로 바꾸는 것이 service로직에 집중을 더 할 수 있을 것 같은데 의견 부탁드립니다 :)

Esunsin commented on src/main/java/com/sparta/hmpah/service/CommentService.java:
```diff
@@ -1,15 +1,140 @@
+    return false;
+  }
+
+  public boolean validateUsername(Comment comment, User userDetails) {
```
이런 부분은 private가 어떤지 의견 물어보고 싶습니다.

LeeJaeHyung committed:
:sparkles:CommentController에서 UserDetailsImpl을 User로 변경하여 CommentService에 전달
-> Service단에서는 좀더 비즈니스 로직에 집중이 가능함

CommentService에 내부에서 이용하는 메서드들의 접근제한자를 private로 변경 -> 좀 더 안정적인 로직으로 전환

LeeJaeHyung committed:
Merge branch 'dev' into feat/comment
